src/server/common_services/auth_server.js

```diff
@@ -669,9 +669,9 @@ function _get_auth_info(account, system, authorized_by, role, extra) {
  * @param {string} bucket_path s3 bucket path (must start from "/")
  * @returns  {Promise<boolean>} true if the account has permission to perform the action on the bucket
  */
-async function has_bucket_action_permission(bucket, account, action, bucket_path = "") {
+async function has_bucket_action_permission(bucket, account, action, bucket_path, req_query) {
     dbg.log1('has_bucket_action_permission:', bucket.name, account.email, bucket.owner_account.email);
-
+    bucket_path = bucket_path ?? ""; // Default to empty string
```

Comment:
again. why is it needed. you can put req_query before bucket_path

Reply:
done